e2eTests/scenarios/PowerPlatform/runtest.ps1

```diff
@@ -0,0 +1,101 @@
+    [string] $appSourceTemplate = $global:appSourceTemplate,
+    [string] $adminCenterApiToken = ($global:SecureAdminCenterApiToken | Get-PlainText),
+    [string] $licenseFileUrl = ($global:SecureLicenseFileUrl | Get-PlainText)
+)
+
+Write-Host -ForegroundColor Yellow @'
+#  _____                       _____  _       _    __                        _____       _       _   _
+# |  __ \                     |  __ \| |     | |  / _|                      / ____|     | |     | | (_)
+# | |__) |____      _____ _ __| |__) | | __ _| |_| |_ ___  _ __ _ __ ___   | (___   ___ | |_   _| |_ _  ___  _ __
+# |  ___/ _ \ \ /\ / / _ \ '__|  ___/| |/ _` | __|  _/ _ \| '__| '_ ` _ \   \___ \ / _ \| | | | | __| |/ _ \| '_ \
+# | |  | (_) \ V  V /  __/ |  | |    | | (_| | |_| || (_) | |  | | | | | |  ____) | (_) | | |_| | |_| | (_) | | | |
+# |_|   \___/ \_/\_/ \___|_|  |_|    |_|\__,_|\__|_| \___/|_|  |_| |_| |_| |_____/ \___/|_|\__,_|\__|_|\___/|_| |_|
+#
+#
+# This test tests the following scenario:
+#
+#  - For the three bcsamples repositories: microsoft/bcsamples-takeorder, microsoft/bcsamples-CoffeeMR and microsoft/bcsampels-WarehouseHelper do the following:
+#  - Create a new repository with the same content as the repository
+#  - Run the Update AL-Go System Files with the latest AL-Go version (main)
+#  - Run the Update AL-Go System Files with the test version
+#  - Check that PP workflows exists
+#  - Run the "CI/CD" workflow
+#  - Check that artifacts are created for the app and the PowerPlatform solution
+#  - Remove the repository
+#
+'@
+
+$errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
+
+Remove-Module e2eTestHelper -ErrorAction SilentlyContinue
+Import-Module (Join-Path $PSScriptRoot "..\..\e2eTestHelper.psm1") -DisableNameChecking
+
+$branch = "main"
+$template = "https://github.com/$pteTemplate"
+$repoPath = (Get-Location).Path
+
+foreach($sourceRepo in @('bcsamples-takeorder', 'bcsamples-CoffeeMR','bcsamples-WarehouseHelper')) {
+    Push-Location
+    $repository = "$githubOwner/$repoName-$($sourceRepo.Split('-')[1])"
+
+    # Login
+    SetTokenAndRepository -github:$github -githubOwner $githubOwner -token $token -repository $repository
+
+    # Create repository1
+    CreateAlGoRepository `
+        -github:$github `
+        -template "https://github.com/microsoft/$sourceRepo" `
+        -repository $repository `
+        -branch $branch `
+        -contentScript {
+            Param([string] $path)
+            Remove-PropertiesFromJsonFile -path (Join-Path $path '.github/AL-Go-Settings.json') -properties @('environments','DeployTo*')
+        }
+
+    $repoPath = (Get-Location).Path
+    Write-Host "Repo Path: $repoPath"
+
+    $settings = Get-Content -Path '.github/AL-Go-Settings.json' -Raw | ConvertFrom-Json
+    Write-Host "PowerPlatform Solution Folder: $($settings.powerPlatformSolutionFolder)"
+
+    if ($settings.templateUrl -eq 'https://github.com/Microsoft/AL-Go-PTE@PPPreview') {
```

Comment:
Why is this something we're testing? 🤔 Shouldn't we just update the bcsamples repos to use AL-Go-PTE@preview/main whenever this PR merges?

Reply:
It is the hen and egg problem - tests will fail until we have done so and we cannot update the sample repos until we have deployed to real preview.
So... - having the tests be resilient means that I can remove this section after we merge

Reply:
Either way, this needs to be cleaned up after this PR is merged. How do we track this clean up?

Reply:
IMO - It is something we can clean up any time later really when we revisit this test - no big deal, but I can create an issue if you redeem necessary